src/client/sandbox/node/live-node-list/wrapper-base.js

```diff
@@ -0,0 +1,36 @@
+const DEFINED_PROPERTIES_COUNT = 10000;
+
+export class LiveNodeListWrapperBase {
+    constructor () {
+        const defineProperty = function (index, isEnumerable) {
+            Object.defineProperty(this, index, {
+                enumerable:   isEnumerable,
+                configurable: true,
+
+                get: function () {
+                    this._refreshNodeList();
+
+                    return this._filteredNodeList[index];
+                }
+            });
+        };
+
+        for (let i = 0; i < DEFINED_PROPERTIES_COUNT; i++)
+            defineProperty.call(this, i, false);
+
+        Object.defineProperty(this, '_defineProperty', { value: defineProperty });
+        Object.defineProperty(this, '_refreshNodeList', {
+            value: () => {
+                throw new Error('Not implemented');
+            }
+        });
+    }
+}
+
+LiveNodeListWrapperBase.prototype = NodeList.prototype;
+
+export const NODE_LIST_WRAPPER = new LiveNodeListWrapperBase();
```

Comment:
> usually we use UPPER_CASE for string/numeric consts or objects with string/numeric consts.

Not only. We use upper case notation for object of any type that uses as constant or as etalon object.
For example, see https://github.com/DevExpress/testcafe-hammerhead/blob/master/src/client/utils/dom.js#L32

`NODE_LIST_WRAPPER`, `HTML_COLLECTION_WRAPPER` are etalons for construct live node list collection of an appropriate type.